File: WebAPI/FootballCompetitionManagerWebApi/Models/Partials/LeagueSeason.cs
using FootballCompetitionManagerWebApi.Models.DTOs;

namespace FootballCompetitionManagerWebApi.Models
{
    public partial class LeagueSeason
    {
        public LeagueSeason() { }

        public LeagueSeason(LeagueSeasonDTO leagueSeasonDto)
        {
            LeagueSeasonId = leagueSeasonDto.LeagueSeasonId;
            CompetitionId = leagueSeasonDto.Competition.CompetitionId;
            DoubleMatches = leagueSeasonDto.DoubleMatches;
            Competition = new Competition(leagueSeasonDto.Competition);
        }
    }
}

File: Android/app/src/main/java/pl/pwr/footballcompetitionmanager/model/LeagueSeason.kt
package pl.pwr.footballcompetitionmanager.model

import com.google.gson.annotations.SerializedName

class LeagueSeason(@SerializedName("LeagueSeasonId") val leagueSeasonId: Int?,
                   @SerializedName("DoubleMatches") var doubleMatches: Boolean,
                   @SerializedName("Competition") val competition: Competition) {

    constructor(doubleMatches: Boolean, competition: Competition)
            : this(null, doubleMatches, competition)
}